### DISCUSSION TITLE
Selection points by mouse with "plotter.enable_rubber_band_syle" or custom interactorstyle

### DISCUSSION THREAD

**Discussion by Gerd-RVHG** [pyvista/pyvista#3323] [Q&A]
Hello all,
i want to select points by mouse. I found no example :-(. The idea is to use "plotter.enanble_rubber_band_style" or a custom interactorstyle. How can i get the position of the selection rectangle and add for examble a callback that i can highlight the points. Have somebody an example for this. 
Thanks

**Comment by banesullivan** [comment]
Have you tried `enable_cell_picking()` or `enable_point_picking()`:

- https://docs.pyvista.org/api/plotting/_autosummary/pyvista.Plotter.enable_cell_picking.html
- https://docs.pyvista.org/api/plotting/_autosummary/pyvista.Plotter.enable_point_picking.html

**Comment by Gerd-RVHG** [comment]
Thanks a lot of your answer. Yes, i have tried both functions. The problem is that you can pick only one point/cell. With the callback, i can do my addionally function.I want to pick multiple points. With the function "enable_rubber_band_style" (https://docs.pyvista.org/api/plotting/_autosummary/pyvista.Plotter.enable_rubber_band_style.html) i can create a Selektion recangle. This function does not have any callback. The question is, how can i get the coordinates of rectangle? And how can i use this function in combination with a vtkAreaPicker (like described  in the documentation)? Am 14.09.2022 15:12 schrieb Bane Sullivan ***@***.***>:
Have you tried enable_cell_picking() or enable_point_picking():
https://docs.pyvista.org/api/plotting/_autosummary/pyvista.Plotter.enable_cell_picking.htmlhttps://docs.pyvista.org/api/plotting/_autosummary/pyvista.Plotter.enable_point_picking.html

**Comment by koenterheegde0507** [comment]
Hey, did you already solve this? I also want to get the coordinates of the drawn rectangle so I can implement it in `pyvista.DataSetFilters.clip_box` as the x and y coordinates for the bounds argument. @banesullivan would you have any other idea to clip parts of a mesh by drawing a rectangle and remove everything thats inside it?
